src/ApiClient.js

```diff
@@ -90,6 +90,10 @@ class ApiClient {
         // else send back the parsed error
         .then(parsedErr => { throw parsedErr })
     }
+    // fix of "Unexpected end of JSON input" when HTTP response is 204 No Content
+    if (response.status === 204) {
+      return { json: () => () => null }
+    }
```

Comment:
Hey, this is great, thanks!

This won't work for clients who aren't using the default json format. We need to generalize this so it handles all of the default formats (arrayBuffer, blob, formData, json, or text). I'd also need to look through https://developer.mozilla.org/en-US/docs/Web/API/Response to make sure we're not breaking any other fields by mocking the object like this.

I can do some thinking on this and push a change next week, but if you're able to get it done before then I'll happily merge it once it's ready 😄